cardano-db-sync/src/Cardano/DbSync/Era/Universal/Block.hs

```diff
@@ -64,16 +65,18 @@ insertBlockUniversal ::
   ApplyResult ->
   ReaderT SqlBackend m (Either SyncNodeError ())
 insertBlockUniversal syncEnv shouldLog withinTwoMins withinHalfHour blk details isMember applyResult = do
+  -- if we're syncing within 2 mins of the tip, we optimise the caches.
+  newCache <- if isSyncedWithintwoMinutes details then optimiseCaches cache else pure cache
```

Comment:
I just realised, the CacheStatus, is not the actual state of the cache, so changing the isOptimised flag here, will not stay in the `syncEnv` for future blocks.